src/OpenTelemetry.Extensions/Trace/RateLimitingSampler.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Based on the jaeger remote sampler for Java from https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/RateLimitingSampler.java
+
+using System.Globalization;
+using OpenTelemetry.Extensions.Internal;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry;
+
+/// <summary>
+/// Rate limiting sampler that can be used to sample traces at a constant rate.
+/// </summary>
+public class RateLimitingSampler : Sampler
+{
+    private const string SAMPLERTYPE = "ratelimiting";
+    private const string SAMPLERTYPEKEY = "sampler.type";
+    private const string SAMPLERPARAMKEY = "sampler.param";
+
+    private readonly RateLimiter rateLimiter;
+    private readonly SamplingResult onSamplingResult;
+    private readonly SamplingResult offSamplingResult;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RateLimitingSampler"/> class.
+    /// </summary>
+    /// <param name="maxTracesPerSecond">The maximum number of traces that will be emitted each second.</param>
+    public RateLimitingSampler(int maxTracesPerSecond)
+    {
+        double maxBalance = maxTracesPerSecond < 1.0 ? 1.0 : maxTracesPerSecond;
+        this.rateLimiter = new RateLimiter(maxTracesPerSecond, maxBalance);
```

Comment:
I don't see how this works, given a maxTracesPerSecond of >= 1, both maxTracesPerSeconf and maxBalance will always be the same. Is that intended?

Reply:
If the maxBalance is less than 1, and we use a weight of 1, there will never be enough of a balance to pass anything. Using the sample rate as the maxBalance provides a good default - the first N items will always be passed, after that they will be sampled out until enough balance is available. The maxBalance provides a cap for the case that there are pauses in the data - if there is no data for multiple seconds, it doesn't enable a large balance to build up.